scripts/cibuild

```diff
@@ -18,16 +18,22 @@ if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
     if [ "${1:-}" = "--help" ]; then
         usage
     else
+        CORES=${CORES:=3}
+        # Strip the patch number off of GDAL_VERSION for tagging
+        GDAL_MAJOR_VERSION=$(echo "${GDAL_VERSION}" | cut -d "." -f 1,2)
+        GDAL_MAJOR_VERSION_FIRST=$(echo "${GDAL_VERSION}" | cut -d "." -f 1)
+        if [[ $GDAL_MAJOR_VERSION_FIRST == "2" ]]; then
+            LIBPROJ_VERSION=${LIBPROJ_VERSION:=4.9.3}
+        fi
+        LIBPROJ_VERSION=${LIBPROJ_VERSION:=7.1.0}
+
```

Comment:
Our practice is to define dependency versions in the matrix.

```diff
diff --git a/.travis.yml b/.travis.yml
index 7f51280..091f03f 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -2,10 +2,10 @@ language: bash
 services: docker

 env:
-  - GDAL_VERSION=2.4.3 OPENJDK_VERSION=8 OPENJPEG_VERSION=2.3.0 VARIANT=slim
-  - GDAL_VERSION=2.4.3 OPENJDK_VERSION=11 OPENJPEG_VERSION=2.3.0 VARIANT=slim
-  - GDAL_VERSION=3.1.2 OPENJDK_VERSION=8 OPENJPEG_VERSION=2.3.0 VARIANT=slim
-  - GDAL_VERSION=3.1.2 OPENJDK_VERSION=11 OPENJPEG_VERSION=2.3.0 VARIANT=slim
+  - GDAL_VERSION=2.4.3 OPENJDK_VERSION=8 OPENJPEG_VERSION=2.3.0 LIBPROJ_VERSION=4.9.3 VARIANT=slim
+  - GDAL_VERSION=2.4.3 OPENJDK_VERSION=11 OPENJPEG_VERSION=2.3.0 LIBPROJ_VERSION=4.9.3 VARIANT=slim
+  - GDAL_VERSION=3.1.2 OPENJDK_VERSION=8 OPENJPEG_VERSION=2.3.0 LIBPROJ_VERSION=7.1.0 VARIANT=slim
+  - GDAL_VERSION=3.1.2 OPENJDK_VERSION=11 OPENJPEG_VERSION=2.3.0 LIBPROJ_VERSION=7.1.0 VARIANT=slim

 script:
   - ./scripts/cibuild
```

We shouldn't define some versions in this script, and other versions in the matrix, we should go all in one a single approach.

Reply:
@rbreslow btw so there should be no sainity checks even if someone tries to build GDAL 3 with PROJ 4?

Reply:
Correct.

The intended purpose of this repository is to build artifacts, with parameters defined by the matrix, for others to consume. Not for folks to check out the repository and build their own permutations (although there's nothing stopping that).

There isn't a sanity check if someone tries to build GDAL 3 with an invalid version of OpenJDK or OpenJPEG.

Reply:
@rbreslow :+1: sounds good to me